### PR TITLE
Add MethodOfLines Benchmark

### PR DESCRIPTION
**Pull request by xtalax**
Adds a benchmark for [MethodOfLines.jl](https://github.com/SciML/MethodOfLines.jl), using Burgers Equation.

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
@staticfloat or @anandijain , the hash thing is back?

**Comment by staticfloat** [comment]
I fixed a couple problems in https://github.com/SciML/SciMLBenchmarks.jl/pull/426, merge that then rebase this on top of `master`.

File: benchmarks/MOLPDE/burgers_fdm_wpd.jmd [hunk 1]
```diff
@@ -48,17 +49,76 @@ function burgers(N, ϵ)
...
+    x_max = dx*N
+    t_max = 1.0
+    
+    xs = dx:dx:x_max
+
+    eq = Dt(u(x, t)) ~ ϵ * Dxx(u(x, t)) - u(x, t) * Dx(u(x, t))
+
+    bcs = [u(x,0.0) ~ exp(-(x - μ0)^2 / (2 * σ0^2)),
+           u(dx,t) ~ 0,
+           d(dx*N,t) ~ 0]
```
Comment (MilkshakeForReal) [inline benchmarks/MOLPDE/burgers_fdm_wpd.jmd:76]: `u(dx*N,t)~0`

File: benchmarks/MOLPDE/burgers_fdm_wpd.jmd [hunk 2]
```diff
@@ -48,17 +49,76 @@ function burgers(N, ϵ)
...
+# Time discretization with MethodOfLines
+println("Discretization time wih MethodOfLines.jl:")
+@time xs, prob_mol = burgers_mol(512, 1e-3)
+
+# Solve with MOL
+sol_mol = solve(prob, Vern9(); abstol=1e-14, reltol=1e-14)
+test_sol_mol = TestSolution(sol_mol);
+
 tslices = [0.0 0.25 0.50 0.75 1.00]
+labels = ["t = $t" for t in tslices
```
Comment (MilkshakeForReal) [inline benchmarks/MOLPDE/burgers_fdm_wpd.jmd:111]: `labels = ["t = $t" for t in tslices]`

**Comment by ChrisRackauckas** [comment]
Let's get this updated. See https://benchmarks.sciml.ai/dev/StiffODE/Bruss/. We can now have multiple problems and allow for a `:prob_choice` in the benchmark. It would be interesting to update this to be alternative prob choices on the FDM ones and see how MOL.jl performs against the hand-written discretizations.

**Comment by xtalax** [comment]
I already have the working bruss code, let me see how to integrate this

**Comment by xtalax** [comment]
I expect `WorkPrecisionSet`  will need special handling for pdesol, I'm just passing the original_sol at the moment. I will duplicate all the MTKitize algs as that's the most equivalent

**Comment by xtalax** [comment]
@ChrisRackauckas  how is bruss looking now? And will unused prob kwargs be passed down to the solve, or do I need to do something special?

**Comment by ChrisRackauckas** [comment]
Add it as a 4th version in Bruss. Don't delete the fastprob one.

**Comment by xtalax** [comment]
@ChrisRackauckas looking at the wp setups, there are cases where `:prob_choice` is 2 for 2 consecutive identical setups. Should these duplicated ones be changed to 3?

**Comment by ChrisRackauckas** [comment]
Most of the different problem checking is done in just the single test. That is because that's for problem formulation, while the other tests of solvers. However, there is the confounding fact that GMRES benefits a lot less from having a fast symbolic Jacobian definition than the others (because it doesn't use it), and so the balance between solvers can be dependent on the problem formulation. For that reason, there's some cases where the modelingtoolkit version is included in the solver tests to showcase this effect, and notably that uses `jac=true, sparse=true`, and that's why you see a major difference with that included (specifically with the KLU version).

**Comment by ChrisRackauckas** [comment]
Failures look like typos

**Comment by xtalax** [comment]
Looks like the fails were caused by MethodOfLines not being in the manifest, I have instantiatec

**Comment by ChrisRackauckas** [comment]
Rebase for v1.8

**Comment by xtalax** [comment]
Rebased

**Comment by ChrisRackauckas** [comment]
need to update the manifest

**Comment by ChrisRackauckas** [comment]
In the manifest:

> julia_version = "1.8.1"

**Comment by xtalax** [comment]
You mean this should be 1.8.2, or omitted? There are examples of both styles across the package

**Comment by ChrisRackauckas** [comment]
it should be v1.8.2 to match the CI machine. I think that's why it's greyed out and I cannot unblock the runs.

**Comment by ChrisRackauckas** [comment]
There's failures in a good chunk of these.

**Comment by xtalax** [comment]
Does WPSet pass down kwargs to the solve? Need to set `wrap=Val(false)`

**Comment by ChrisRackauckas** [comment]
It does. If it's a problem-specific thing, then the setup will too.

File: benchmarks/StiffODE/Bruss.jmd [hunk 3]
```diff
@@ -357,7 +413,7 @@ setups = [
           ]
 names = ["KenCarp47 KLU MTK", "KenCarp47 GMRES MTK",
          "FBDF KLU MTK", "FBDF GMRES MTK",
-         "CVODE MTK KLU", "CVODE iLU MTK GMRES"
```
Comment (ChrisRackauckas) [inline benchmarks/StiffODE/Bruss.jmd:360]: Why this change?

File: benchmarks/StiffODE/Bruss.jmd [hunk 4]
```diff
@@ -400,7 +456,7 @@ setups = [
 names = ["KenCarp47 KLU MTK", "KenCarp47 GMRES MTK",
          "FBDF KLU MTK", "FBDF GMRES MTK",
          "Rodas5P GMRES MTK",
-         "CVODE MTK KLU", "CVODE iLU MTK GMRES"
```
Comment (ChrisRackauckas) [inline benchmarks/StiffODE/Bruss.jmd:403]: why this change?

Comment (xtalax) [reply benchmarks/StiffODE/Bruss.jmd:403]: Because that setup doesn't reference problem 2

File: benchmarks/StiffODE/Bruss.jmd (hunk 3 above)
Comment (ChrisRackauckas) [reply benchmarks/StiffODE/Bruss.jmd:360]: change the prob choice not the title here.

**Comment by ChrisRackauckas** [comment]
Builds failed.

**Comment by ChrisRackauckas** [comment]
@xtalax status here?

**Comment by xtalax** [comment]
Status is that I can't work out why this is still failing, could copy out the code and project and try to run manually to debug

**Comment by xtalax** [comment]
@ChrisRackauckas this should be done now

**Comment by ChrisRackauckas** [comment]
Okay let's continue the MethodOfLinesPDE set in a later PR
